components/ParameterForm.vue

```diff
@@ -148,13 +148,14 @@ const appStore = useAppStore();
 const paramMetadata = computed(() => appStore.metadata?.parameters);
 
 const formData = ref(
-  // Create a new object with keys set to the id values of the metadata.parameters array of objects, and all values set to default values.
+  // Initialize formData as a dictionary of parameters with values set to defaults if available,
+  // or to the previous scenario's values if any.
   paramMetadata.value?.reduce((acc, { id, defaultOption, options, updateNumericFrom }) => {
-    if (options && !updateNumericFrom) { // Excludes fields whose values depend on others'; we'll set these once all the defaults are set.
+    if (!acc[id] && options && !updateNumericFrom) { // Excludes fields whose values depend on others'; we'll set these once all the defaults are set.
       acc[id] = (defaultOption || options[0].id).toString();
     }
     return acc;
-  }, {} as { [key: string]: string }),
+  }, { ...appStore.currentScenario?.parameters } as { [key: string]: string }),
```

Comment:
I think this two-stage initialisation (with conditional based on the accumulator we're currently accumulating) is quite difficult to read. 
Can't we just initialise fully from the current scenario if there is one, rather than getting the reduce involved at the level of each parameter? I'd also suggest pulling out the initialise from defaults into a separate method to make the basic initialisation logic clearer
```
const initialiseFormDataFromDefaults = () => {
  return paramMetadata.value?.reduce((acc, { id, defaultOption, options, updateNumericFrom }) => {
    if (options && !updateNumericFrom) { // Excludes fields whose values depend on others'; we'll set these once all the defaults are set.
      acc[id] = (defaultOption || options[0].id).toString();
    }
    return acc;
  }, {} as { [key: string]: string })
}
const formData = ref(
  // Initialize formData as a dictionary of parameters with values set to defaults if available,
  // or to the previous scenario's values if any.
  appStore.currentScenario ? {...appStore.currentScenario.parameters} : initialiseFormDataFromDefaults()
);
```

Reply:
Yep, that works, thanks!